awx/main/models/credential/injectors.py

```diff
@@ -39,6 +39,20 @@ def gce(cred, env, private_data_dir):
     f.close()
     os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
     env['GCE_CREDENTIALS_FILE_PATH'] = path
+
+    handle, path = tempfile.mkstemp(dir=private_data_dir)
+    f = os.fdopen(handle, 'w')
+    json.dump(json_cred, f, indent=2)
+    f.close()
+    os.chmod(path, stat.S_IRUSR | stat.S_IWUSR)
+    env['GCP_SERVICE_ACCOUNT_FILE'] = path
```

Comment:
Couldn't you point this env var to the existing file, since it's the same? Maybe this was necessary because the tests were failing. If that's the case, I can come back and work on the tests. The point of the tests is to help enforce (and keep track of) a sensible file & environment structure.

Reply:
Yeah, this was very much to satisfy the tests.

Since "GCP_SERVICE_ACCOUNT_FILE" and "GCE_SERVICE_ACCOUNT_PATH" pointed to the same file, the tests removed "GCE_SERVICE_ACCOUNT_PATH" as a valid environment variable. The tests were very confused when it wasn't expecting GCE_SERVICE_ACCOUNT_PATH to exist, but it did.